pkg/datapath/linux/config/config.go

```diff
@@ -178,10 +177,6 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 	cDefinesMap["CT_TAIL_CALL_BUFFER4"] = "cilium_tail_call_buffer4"
 	cDefinesMap["CT_TAIL_CALL_BUFFER6"] = "cilium_tail_call_buffer6"
 
-	if option.Config.DatapathMode == datapathOption.DatapathModeIpvlan {
-		cDefinesMap["ENABLE_EXTRA_HOST_DEV"] = "1"
```

Comment:
It looks like this is the only place where `ENABLE_EXTRA_HOST_DEV` is defined, so this allows us to drop the `#else` part her in `bpf/bpf_host.c` as well:

https://github.com/cilium/cilium/blob/8f606bafb696076609a01f88204d895576043950/bpf/bpf_host.c#L446-L453